### PR TITLE
Compilation improvements

### PR DESCRIPTION
**Pull request by MasloMaslane**

### CODE REVIEW

File: src/sinol_make/commands/run/__init__.py [hunk 1]
```diff
@@ -191,8 +193,14 @@ def compile(self, solution):
             self.COMPILATION_DIR, "%s.compile_log" % self.extract_file_name(solution))
         source_file = os.path.join(os.getcwd(), "prog", self.get_solution_from_exe(solution))
         output = os.path.join(self.EXECUTABLES_DIR, self.get_executable(solution))
+
+        weak_compilation_flags = False
+        if hasattr(self, "args") and self.args.weak_compilation_flags:
+            weak_compilation_flags = True
```
Comment (tonowak) [inline src/sinol_make/commands/run/__init__.py:199]:
```suggestion
        weak_compilation_flags = hasattr(self, "args") and self.args.weak_compilation_flags
```

Comment (tonowak) [reply src/sinol_make/commands/run/__init__.py:199]: Also, can `self` not have `args`?

**Review by tonowak — COMMENTED** [review]
Looks good, my only suggestions:

File: src/sinol_make/commands/run/__init__.py (hunk 1 above)
Comment (MasloMaslane) [reply src/sinol_make/commands/run/__init__.py]: Changed 7d050b88495eb66ddda18314deefaeca167f4b01

**Review by tonowak — COMMENTED** [review]
Are there any tests? E.g. ones that check whether the new sinol-make flag indeed compiles some codes that fail to compile on full flags.

**Comment by MasloMaslane** [comment]
No, I'll add one

**Review by tonowak — APPROVED** [review]
Nice, let's release a new sinol-make version!

**Pull request merged**